packages/react/src/components/Stepper/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import TooltipTabstop from '../TooltipTabstop';
+
+interface BaseStepProps {
+  status: 'current' | 'complete' | 'future';
+  className?: string;
+}
+
+interface StepWithChildren extends BaseStepProps {
+  children: React.ReactNode;
+  tooltip?: never;
```

Comment:
@scurker I forgot to circle back and try out your suggestion here for replacing `never` here with something like: 

```
function propsWithTooltip(props: StepWithChildren | StepWithTooltip): props is StepWithTooltip {
    return !!(props as StepWithTooltip).tooltip
}
```